src/Filesystem.cpp

```diff
@@ -128,7 +128,7 @@ StringList Filesystem::searchPath() const
  *
  * \note	This function will also return the names of any directories in a specified search path
  */
-StringList Filesystem::directoryList(const std::string& dir, const std::string& filter) const
+StringList Filesystem::directoryList(const std::string& dir, const std::string& filter /*= std::string{}*/) const
```

Comment:
I have a mild preference to not include the comment here. Comments have a tendency to grow stale. Plus I prefer to not repeat information. Repeated information tends to increase maintenance costs, or allows for skew when not all values are updated together.